src/lib/snark_profiler_lib/snark_profiler_lib.ml

```diff
@@ -494,113 +521,91 @@ let create_ledger_and_zkapps ?(min_num_updates = 1) ?(num_proof_updates = 0)
-            | Signature _ ->
-                (pc, p :: sc, na, nonce)
-            | _ ->
-                (pc, sc, p :: na, nonce) )
-      in
-      printf
-        !"\n\n\
-          Generated zkapp transactions with %d updates and %d proof updates in \
-          %f secs\n\
-          %!"
-        (List.length other_parties + 1)
-        (List.length proof_parties)
-        Time.(Span.to_sec (diff (now ()) start)) ;
-      let%bind.Async.Deferred permutations =
-        permute proof_parties (signature_parties @ no_auths) [] []
-        |> Async.Deferred.List.filter_mapi ~how:`Sequential
-             ~f:(fun i (account_updates : Account_update.Simple.t list) ->
-               let p =
-                 Zkapp_command.of_simple { simple_parties with account_updates }
-               in
-               let combination =
-                 Transaction_key.of_zkapp_command ~constraint_constants ~ledger
-                   p
-               in
-               let perm_string =
-                 List.fold ~init:"S" account_updates
-                   ~f:(fun acc (p : Account_update.Simple.t) ->
-                     match p.authorization with
-                     | Proof _ ->
-                         acc ^ "P"
-                     | Signature _ ->
-                         acc ^ "S"
-                     | None_given ->
-                         acc ^ "N" )
-               in
-               if Transaction_key.Table.mem transaction_combinations combination
-               then (
-                 printf "Skipping %s\n%!" perm_string ;
-                 Async.Deferred.return None )
-               else (
-                 printf
-                   !"Generated updates permutation %d: %s\n\
-                     Updating authorizations...\n\
-                     %!"
-                   i perm_string ;
-                 (*Update the authorizations*)
-                 let%map.Async.Deferred p =
-                   Zkapp_command_builder.replace_authorizations ~prover ~keymap
-                     p
-                 in
-                 Transaction_key.Table.add_exn transaction_combinations
-                   ~key:combination
-                   ~data:(p, Time_values.empty, perm_string) ;
-                 Some p ) )
-      in
-      generate_zkapp ~num_proof_updates:(num_proof_updates + 1) ~num_updates
-        (permutations @ acc) nonce
+  let generate_command_of_constitution constitution nonce =
+    let start = Time.now () in
+    let num_updates = Constitution.num_updates_needed constitution - 1 in
+    let num_proof_updates = constitution.proof_segments in
+    let empty_sender, spec = test_spec nonce ~num_proof_updates ~num_updates in
+    let%bind.Async.Deferred parties =
+      Transaction_snark.For_tests.update_states ~zkapp_prover_and_vk
+        ~constraint_constants ~empty_sender spec
+        ~receiver_auth:Control.Tag.Signature
+    in
+    let simple_parties = Zkapp_command.to_simple parties in
```

Comment:
What's the name `parties` / `simple_parties` / `other_parties` here indicating?

Reply:
"Party" here is supposed to mean "account involved in the transaction". I guess that naming scheme was chosen because this command generator only cares about the account updates in the commands. Really, it's:

- `parties` - zkApp transaction as a `Zkapp_command.t`
- `simple_parties` - zkApp transaction as a `Zkapp_command.Simple.t` 
- `other_parties` - account updates (as `Account_update.Simple.t`) in the transaction aside from the fee payer update

etc.